projects/common/src/time/time-range.service.ts

```diff
@@ -62,16 +66,44 @@ export class TimeRangeService {
     this.setTimeRange(this.getCurrentTimeRange());
   }
 
+  private globalTimeRangeInitialization(): Observable<TimeRange> {
+    return this.navigationService.navigation$.pipe(
+      take(1), // Wait for first navigation
+      switchMap(activatedRoute => activatedRoute.queryParamMap), // Get the params from it
+      take(1), // Only the first set of params
+      map(paramMap => paramMap.get(TimeRangeService.TIME_RANGE_QUERY_PARAM)), // Extract the time range value from it
+      filter((timeRangeString): timeRangeString is string => !isEmpty(timeRangeString)), // Only valid time ranges
+      map(timeRangeString => this.timeRangeFromUrlString(timeRangeString)),
+      catchError(() => EMPTY)
+    );
+  }
+
+  private pageTimeRangeInitialization(): Observable<TimeRange> {
+    return this.navigationService.navigation$.pipe(
+      filter(
+        activatedRoute => !isNil(activatedRoute.snapshot.queryParamMap.get(TimeRangeService.TIME_RANGE_QUERY_PARAM))
+      ),
+      map(activeRoute => {
+        const timeRangeQueryParamString = activeRoute.snapshot.queryParamMap.get(
+          TimeRangeService.TIME_RANGE_QUERY_PARAM
+        );
+
+        return this.timeRangeFromUrlString(timeRangeQueryParamString!);
+      })
+    );
+  }
+
   private initializeTimeRange(): void {
-    this.navigationService.navigation$
+    this.featureStateResolver
+      .getFeatureState(ApplicationFeature.PageTimeRange)
       .pipe(
-        take(1), // Wait for first navigation
-        switchMap(activatedRoute => activatedRoute.queryParamMap), // Get the params from it
-        take(1), // Only the first set of params
-        map(paramMap => paramMap.get(TimeRangeService.TIME_RANGE_QUERY_PARAM)), // Extract the time range value from it
-        filter((timeRangeString): timeRangeString is string => !isEmpty(timeRangeString)), // Only valid time ranges
-        map(timeRangeString => this.timeRangeFromUrlString(timeRangeString)),
-        catchError(() => EMPTY)
+        switchMap(featureState => {
+          if (featureState === FeatureState.Enabled) {
+            return this.pageTimeRangeInitialization();
```

Comment:
nit - the method returns an observable of a time range, doesn't actually do the initialization and should be renamed (same with the global version).